api.visual/src/org/netbeans/api/visual/model/ObjectScene.java

```diff
@@ -534,4 +588,20 @@ public void widgetHovered (Widget widget) {
 
     }
 
+    /**
+     * Finds the state for the given object. The object must be a valid part of the model, although
+     * it may not be registered yet and no widgets are created for it. The method may throw {@link IllegalArgumentException}
+     * for instances that are not proper models for the scene. The default method returns {@link ObjectState#createNormal}
+     * for all inputs
+     * 
+     * @param o the object
+     * @return the ObjectState
+     * @throws IllegalArgumentException if the object is not valid for the scene.
```

Comment:
Isn't returning `null` easier to use than an exception?

Reply:
Thank for pointing this out, returning null more conforms to the "unknown" handling of the other API mehods. I will fix javadoc.